website/server/controllers/api-v3/groups.js

```diff
@@ -650,15 +650,6 @@ api.joinGroup = {
 
     // Check the inviter again, could be a deleted account
     if (inviter) {
```

Comment:
If I'm not missing something: The inviter is currently always loaded. However, it is only actually used for party invitations. This means, this could be refactored a bit to move the check for the group being a party "outside" of even loading the inviter. So that for invites to private guilds, the inviter is not loaded since it's not updated/used anyway.

Reply:
Shuffled some things around!